src/main/resources/Confluence/Macros/MicrosoftStream.xml

```diff
@@ -0,0 +1,1039 @@
+    <property>
+      <defaultValue>300px</defaultValue>
+    </property>
+    <property>
+      <description>The viewer height. Can be specified either in pixels (for example, 400px) or as a percentage of the available page height (for example, 50%).</description>
+    </property>
+    <property>
+      <mandatory>0</mandatory>
+    </property>
+    <property>
+      <name>height</name>
+    </property>
+    <property>
+      <type/>
+    </property>
+  </object>
+  <object>
+    <name>Confluence.Macros.MicrosoftStream</name>
+    <number>3</number>
+    <className>XWiki.WikiMacroParameterClass</className>
+    <guid>6b69fdec-ac17-4bb6-ad5d-9212ec66bb9b</guid>
+    <class>
+      <name>XWiki.WikiMacroParameterClass</name>
+      <customClass/>
+      <customMapping/>
+      <defaultViewSheet/>
+      <defaultEditSheet/>
+      <defaultWeb/>
+      <nameField/>
+      <validationScript/>
+      <defaultValue>
+        <disabled>0</disabled>
+        <name>defaultValue</name>
+        <number>4</number>
+        <prettyName>Parameter default value</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </defaultValue>
+      <description>
+        <disabled>0</disabled>
+        <name>description</name>
+        <number>2</number>
+        <prettyName>Parameter description</prettyName>
+        <rows>5</rows>
+        <size>40</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.TextAreaClass</classType>
+      </description>
+      <mandatory>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>mandatory</name>
+        <number>3</number>
+        <prettyName>Parameter mandatory</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </mandatory>
+      <name>
+        <disabled>0</disabled>
+        <name>name</name>
+        <number>1</number>
+        <prettyName>Parameter name</prettyName>
+        <size>30</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <type>
+        <disabled>0</disabled>
+        <name>type</name>
+        <number>5</number>
+        <prettyName>Parameter type</prettyName>
+        <size>60</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </type>
+    </class>
+    <property>
+      <defaultValue/>
+    </property>
+    <property>
+      <description>Position of the video relatively to its container. Accepted values are left, center and right.</description>
+    </property>
+    <property>
+      <mandatory>0</mandatory>
+    </property>
+    <property>
+      <name>alignment</name>
+    </property>
+    <property>
+      <type/>
```

Comment:
I worked on this but it doesn't work on the minimal supported version 11.10 (probably also related to https://jira.xwiki.org/browse/XWIKI-17768, see [the PDFViewer related issue](https://github.com/xwikisas/macro-pdfviewer/issues/28))

So we could wait for the project to upgrade the parent version to 12.10 and add this feature then or add it now and accept that users that are on an older version won't be able to select an alignment (since a select without options is displayed)
A third option would be to move the macro to java, to be able to set a `PropertyDisplayType`, but I don't think it's worth it since we should move to 12.10 pretty soon.

Reply:
After confirming with @silviamacovei, I added the alignment dropdown. This will make the selection of an alignment possible only by syntax for versions where https://jira.xwiki.org/browse/XWIKI-17768 is still present

Reply:
@oanalavinia what exception were you getting? I would have expected a converter between Enum and String to be available (like it happens on Velocity, but maybe that conversion is specific to Velocity).

Reply:
Actually, after further investigations, it does not seem to be related to the issue I specified
So what is not working is `$displayer.type.getEnumConstants()` from https://github.com/xwiki/xwiki-platform/blob/xwiki-platform-11.10/xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/html_displayer/enum/edit.vm#L5
And there is only this warning `WARN  o.x.v.i.DefaultVelocityEngine  - Cannot retrieve method getEnumConstants from object of class java.lang.Class due to security restrictions.` 
On a 12.10.6 instance `#getEnumConstants()` is working and the warning is not displayed anymore (I get another about security restrictions, but linked to other method)